### PR TITLE
Refac: Migrate 3 components from Enzyme to RTL

### PR DESCRIPTION
**Pull request by EshaanAgg**
## Which problem is this PR solving?
Fixes part of #1668 

## Description of the changes
Migrates the following components from Enzyme to RTL
1. `CircularProgressbar`
2. `ErrorMessage`
3. `TraceName`

- Removed the instances of snapshot testing from the same and migrated to tests checking the functionality associated with them
- Refactored some of the original code of the components to make them more readable

## How was this change tested?
- Running the tests locally

## Checklist
- [X] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [X] I have signed all commits
- [X] I have added unit tests for the new functionality
- [X] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js [hunk 1]
```diff
@@ -13,141 +13,124 @@
...
-    wrapper = shallow(<ErrorMessage error={error} />);
-  });
+describe('<ErrorMessage>', () => {
+  const errorMessage = 'some error message';
 
   it('is ok when not passed an error', () => {
-    wrapper.setProps({ error: null });
-    expect(wrapper).toBeDefined();
+    render(<ErrorMessage />);
+    expect(screen).toBeDefined();
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/common/ErrorMessage.test.js:26]: isn't screen always defined?

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js [hunk 2]
```diff
@@ -13,141 +13,124 @@
...
-    expect(msg.shallow().text()).toMatch(error);
+    render(<ErrorMessage error={errorMessage} />);
+    expect(screen.getByText(errorMessage)).toBeInTheDocument();
   });
 
   it('<Details /> renders empty on string error', () => {
-    wrapper = shallow(<Details error={error} />);
-    expect(wrapper).toMatchSnapshot();
+    render(<Details error={errorMessage} />);
+    expect(screen).toBeDefined();
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/common/ErrorMessage.test.js:36]: is this ever false? why not assert by error message?

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js [hunk 3]
```diff
@@ -13,141 +13,124 @@
...
   });
 
   it('<Details /> renders wrapper on wrap', () => {
-    error = {
-      message: 'some-http-ish-message',
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/common/ErrorMessage.test.js:47]: why not validate the message too?

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js [hunk 4]
```diff
@@ -13,141 +13,124 @@
...
+    const error = {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/common/ErrorMessage.test.js:40]: this struct is repeated many times, maybe pull into a global const?

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js [hunk 5]
```diff
@@ -13,141 +13,124 @@
...
-      expect(details.length).toBe(1);
-      const detailsWrapper = details.shallow();
-      expect(detailsWrapper).toMatchSnapshot();
+    // All the keys expect httpBody should be present as is
+    Object.keys(error).forEach(key => {
+      if (key !== 'httpBody') expect(screen.getByText(error[key])).toBeInTheDocument();
     });
+
+    // The body should be truncated
+    expect(screen.getByText(`${error.httpBody.slice(0, 1021).trim()}...`)).toBeInTheDocument();
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/common/ErrorMessage.test.js:119]: what us this magic number 1021? If the component has a specific limit, let's expose that as a constant from the component and use here.

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js (hunk 1 above)
Comment (EshaanAgg) [reply packages/jaeger-ui/src/components/common/ErrorMessage.test.js:26]: My bad. Will fix.

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js (hunk 2 above)
Comment (EshaanAgg) [reply packages/jaeger-ui/src/components/common/ErrorMessage.test.js:36]: Need to do away with the `toBeDefined()` on `screen` as previously pointed. Will fix it.

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js (hunk 3 above)
Comment (EshaanAgg) [reply packages/jaeger-ui/src/components/common/ErrorMessage.test.js:47]: This is a test for the `Details` component, which does not have `message` listed in its props and doesn't render the same. `message` is needed when testing the `Message` or `ErrMessage` component.

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js (hunk 4 above)
Comment (EshaanAgg) [reply packages/jaeger-ui/src/components/common/ErrorMessage.test.js:40]: Tried doing that it causes more redirection and hard-to-read tests than it reduces duplication. The primary reason for the same being each test changes/deletes just one key of the object. 
If we were to do that, then in each test we would first need to restore the global const from previous test, and then augment it again according to it's need.

File: packages/jaeger-ui/src/components/common/ErrorMessage.test.js (hunk 5 above)
Comment (EshaanAgg) [reply packages/jaeger-ui/src/components/common/ErrorMessage.test.js:119]: It's the maximum limit `1024` - length of `...`. Will expose it as a constant.

**Review by yurishkuro — APPROVED** [review]
Thanks!

**Pull request merged**